File: 2022/day01/ruby/solutionA.rb
#!/bin/env ruby
# frozen_string_literal: true

input = File.read('../input.txt')

# @param {String} input
# @return {Integer[][]}
def format(input)
  input.split("\n\n").map! do |numbers|
    numbers.split("\n").map!(&:to_i)
  end
end

# @param {Integer[][]} elves
# @return {Integer}
def solution(elves)
  elves.map!(&:sum).max
end

print solution(format(input))

File: day01/python/solutionA.py
#!/bin/env python

input = open('../input.txt').read()

def format(input: str) -> 'map[map[int]]':
    return map(lambda n: map(int, n.split('\n')), input.split('\n\n'))

def solution(elves: 'map[map[int]]') -> int:
    return max(map(sum, elves))

print(solution(format(input)))
